### PR TITLE
lib: Add path functions

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/lib/path.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+function drop_slashes(path : string): string {
+    // Drop all trailing slashes, but never drop the first character.
+    let pos = path.length;
+    while (pos > 1 && path[pos - 1] == "/")
+        pos -= 1;
+    return pos == path.length ? path : path.substr(0, pos);
```
Comment (cockpituous) [inline pkg/lib/path.ts:25]: These 2 added lines are not executed by any test.

File: pkg/lib/path.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export function dirname(path : string): string {
+    const norm = drop_slashes(path);
+    const pos = norm.lastIndexOf("/");
+    if (pos < 0)
+        return ".";
+    else if (pos == 0)
+        return "/";
```
Comment (cockpituous) [inline pkg/lib/path.ts:34]: This added line is not executed by any test.

File: pkg/lib/path.ts (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/path.ts:25]: These 2 added lines are not executed by any test.

File: pkg/lib/path.ts (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/path.ts:34]: This added line is not executed by any test.

File: pkg/lib/path.ts (hunk 1 above)
Comment (jelly) [reply pkg/lib/path.ts:25]: So hmm this coverage does not account for the qunit coverage right? Feature, Bug ;-)

Would be awesome to merge them but sounds tedious so back to reviewing :)

**Review by jelly — APPROVED** [review]
This is cool stuff! 

Follow ups in machines and files would be nice :+1: 

https://github.com/cockpit-project/cockpit-files/blob/d3abd3a9622066277b8d5f0d92b834551cb6f524/src/common.ts#L55

**Pull request merged**